Mono.Addins/Mono.Addins.Database/AddinDatabase.cs

```diff
@@ -1322,63 +1310,63 @@ void InternalScanFolders2 (IProgressStatus monitor, AddinScanResult scanResult)
 				scanResult.RegenerateAllData = true;
 			}
 			
-			AddinScanner scanner = new AddinScanner (this, scanResult, monitor);
-			try {
+			var updater = new AddinRegistryUpdater (this, scanResult);
 
-				// Check if any of the previously scanned folders has been deleted
+			// Check if any of the previously scanned folders has been deleted
 
-				foreach (string file in Directory.GetFiles (AddinFolderCachePath, "*.data")) {
-					AddinScanFolderInfo folderInfo;
-					bool res = ReadFolderInfo (monitor, file, out folderInfo);
-					bool validForDomain = scanResult.Domain == null || folderInfo.Domain == GlobalDomain || folderInfo.Domain == scanResult.Domain;
-					if (!res || (validForDomain && !fs.DirectoryExists (folderInfo.Folder))) {
-						if (res) {
-							// Folder has been deleted. Remove the add-ins it had.
-							scanner.UpdateDeletedAddins (monitor, folderInfo, scanResult);
-						} else {
-							// Folder info file corrupt. Regenerate all.
-							scanResult.ChangesFound = true;
-							scanResult.RegenerateRelationData = true;
-						}
-
-						if (!scanResult.CheckOnly)
-							SafeDelete (monitor, file);
-						else if (scanResult.ChangesFound)
-							return;
+			foreach (string file in Directory.GetFiles (AddinFolderCachePath, "*.data")) {
```

Comment:
Hmmm, why do we scan the whole folder instead of cache items specific to the VS being used?

Maybe we should modify the addin cache layout to be, in example, on Mac:
`~/Library/Caches/VisualStudio/7.0/<path>/*.data`

That way we only scan the ones in the current instance?

Reply:
Maybe folder data files could be classified by domain, like add-in data files. I'll take a look.

Reply:
But I'll optimize this in another PR